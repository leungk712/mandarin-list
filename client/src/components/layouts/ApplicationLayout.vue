<template>
  <div id="layout--application">
    <ApplicationToolbar />
    <NavigationDrawer />
    <v-main>
      <slot />
    </v-main>
    <ApplicationSnackbar />
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import ApplicationSnackbar from "@/components/ApplicationSnackbar.vue";
import ApplicationToolbar from "@/components/layouts/ApplicationToolbar.vue";
import NavigationDrawer from "@/components/layouts/NavigationDrawer.vue";

@Component({
  name: "ApplicationLayout",
  components: { ApplicationSnackbar, ApplicationToolbar, NavigationDrawer }
})
export default class ApplicationLayout extends Vue {}
</script>
