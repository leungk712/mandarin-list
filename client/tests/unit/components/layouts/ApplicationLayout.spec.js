import { createLocalVue, shallowMount } from "@vue/test-utils";
import ApplicationLayout from "@/components/layouts/ApplicationLayout.vue";
import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";

Vue.use(Vuetify);

const localVue = createLocalVue();
localVue.use(Vuex);

const elem = document.createElement("div");
if (document.body) {
    document.body.appendChild(elem);
};

document.body.setAttribute("data-app", true);

const actions = {};

const mockStore = {};

describe('ApplicationLayout', () => {
    let wrapper;
    const store = new Vuex.Store({
        modules: mockStore
    });
    const mountFunction = (options, store = mockStore) => {
        return shallowMount(ApplicationLayout, {
           attachTo: elem,
           localVue,
           store,
           vuetify: new Vuetify(),
           ...options
        });
    };
    describe("template", () => {
        it("should match snapshot", () => {
            wrapper = mountFunction({}, store);
            expect(wrapper.vm.$el).toMatchSnapshot();
            wrapper.destroy();
        });
    });
});