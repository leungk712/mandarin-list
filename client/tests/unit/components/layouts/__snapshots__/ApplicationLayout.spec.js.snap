// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationLayout template should match snapshot 1`] = `
<div
  id="layout--application"
>
  <applicationtoolbar-stub />
   
  <navigationdrawer-stub />
   
  <v-main-stub
    tag="main"
  />
   
  <applicationsnackbar-stub />
</div>
`;
